src/machine/memory/cache/cache_policy.cpp

```diff
@@ -128,4 +132,37 @@ size_t CachePolicyRAND::select_way_to_evict(size_t row) const {
     UNUSED(row)
     return std::rand() % associativity; // NOLINT(cert-msc50-cpp)
 }
+
+CachePolicyPLRU::CachePolicyPLRU(size_t associativity, size_t set_count)
+    : associativity(associativity), associativityCLog2(std::ceil(log2((float)associativity))) {
+    plru_ptr.resize(set_count);
+    for (auto &row : plru_ptr) {
+        row.resize((2 << associativityCLog2)-1,0); // Initially point to block 0
+    }
+}
+
+void CachePolicyPLRU::update_stats(size_t way, size_t row, bool is_valid) {
+    UNUSED(is_valid)
+    uint32_t plru_idx = 0;
+    auto &row_ptr = plru_ptr.at(row);
+    for (uint32_t i = 0; i < associativityCLog2; i++) {
+        row_ptr[plru_idx] = ((way >> (associativityCLog2 - 1 - i)) & 1) ? 0 : 1;
```

Comment:
Could you add some comments here? I think it is correct, but it is rather hard to read. Maybe a little common subexpression extraction to named variables would help.

Reply:
Sure no problem